### PR TITLE
[Merged by Bors] - feat(Topology): a profinite space has countably many clopens iff it is second countable

### PR DESCRIPTION
**Pull request by dagurtomas**
---
This will be used to prove the existence of countable limits in `LightProfinite` 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Compactness/Compact.lean [hunk 1]
```diff
@@ -551,30 +551,45 @@ theorem exists_subset_nhds_of_isCompact' [Nonempty ι] {V : ι → Set X}
   contradiction
 #align exists_subset_nhds_of_is_compact' exists_subset_nhds_of_isCompact'
 
+lemma eq_finite_iUnion_of_isTopologicalBasis__of_isCompact_open (b : ι → Set X)
```
Comment (jcommelin) [inline Mathlib/Topology/Compactness/Compact.lean:554]:
```suggestion
lemma eq_finite_iUnion_of_isTopologicalBasis_of_isCompact_open (b : ι → Set X)
```

File: Mathlib/Topology/Compactness/Compact.lean [hunk 2]
```diff
@@ -551,30 +551,45 @@ theorem exists_subset_nhds_of_isCompact' [Nonempty ι] {V : ι → Set X}
...
+    intro i hi
+    erw [← Set.iUnion_subtype (fun x : ι => x ∈ t.image f') fun i => b i.1]
+    exact Set.subset_iUnion (fun i : t.image f' => b i) ⟨_, Finset.mem_image_of_mem _ hi⟩
+  · apply Set.iUnion₂_subset
+    rintro i hi
+    obtain ⟨j, -, rfl⟩ := Finset.mem_image.mp hi
+    rw [e]
+    exact Set.subset_iUnion (b ∘ f') j
+
+lemma eq_sUnion_finset_of_isTopologicalBasis__of_isCompact_open (b : Set (Set X))
```
Comment (jcommelin) [inline Mathlib/Topology/Compactness/Compact.lean:575]:
```suggestion
lemma eq_sUnion_finset_of_isTopologicalBasis_of_isCompact_open (b : Set (Set X))
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by dagurtomas** [comment]
bors merge

**Pull request closed without merging**